### PR TITLE
feat(cli): separate errors into 'user errors' and 'internal errors'

### PR DESCRIPTION
**Pull request by johnyob**
# Context

Improves CLI error reporting

# Description

This PR:
1. Introduces an `error.rs` file that defines the CLI's result + error type (anyhow w/ internal constructors such as `UserError`).
2. Updates the rest of the CLI to use this error module.
3. Improves various error messages in the CLI

# Manually testing the PR

Running a selection of commands with erroneous outputs

### CODE REVIEW

File: crates/jstz_cli/src/utils.rs [hunk 1]
```diff
@@ -9,8 +9,8 @@ pub fn from_file_or_id(data_or_file: String) -> String {
 }
 
 fn get_stdin() -> String {
-    let lines: Result<Vec<_>, _> = io::stdin().lines().collect();
-    lines.expect("Can't read from stdin").join("\n")
+    let lines: io::Result<Vec<_>> = io::stdin().lines().collect();
+    lines.expect("Failed to read lines from stdin").join("\n")
```
Comment (sam-finch-tezos) [inline crates/jstz_cli/src/utils.rs:13]: Should we definitely be using expect here? It doesn't really matter in the CLI, but an error might be better.

File: crates/jstz_cli/src/sandbox/daemon.rs [hunk 2]
```diff
@@ -228,7 +230,7 @@ fn start_sandbox(cfg: &Config) -> Result<(OctezThread, OctezThread, OctezThread)
 pub fn main(cfg: &mut Config) -> Result<()> {
     // 1. Check if sandbox is already running
     if cfg.sandbox.is_some() {
-        return Err(anyhow::anyhow!("Sandbox is already running!"));
+        bail_user_error!("Sandbox is already running!");
```
Comment (sam-finch-tezos) [inline crates/jstz_cli/src/sandbox/daemon.rs:233]:
```suggestion
        bail_user_error!("The sandbox is already running!");
```

File: crates/jstz_cli/src/logs/trace.rs [hunk 3]
```diff
@@ -34,8 +34,8 @@ pub async fn exec(
                 }
             }
             Err(err) => {
-                println!("Event source error: {}", err);
                 event_source.close();
+                eprintln!("Event source error, with an error: {}", err);
```
Comment (sam-finch-tezos) [inline crates/jstz_cli/src/logs/trace.rs:38]: Is this really better than the original?

**Review by sam-finch-tezos — APPROVED** [review]
lgtm; a couple of minor sugestions.
I think long term It would be better to have interactive recoveries eg "you are already logged in as {current_alias}, logout {current_alias} and log in as {new_alias}? Y/n".  But lets keep this for now.

**Comment by johnyob** [comment]
> I think long term It would be better to have interactive recoveries eg "you are already logged in as {current_alias}, logout {current_alias} and log in as {new_alias}? Y/n". But lets keep this for now.

Dw this is coming

File: crates/jstz_cli/src/utils.rs (hunk 1 above)
Comment (johnyob) [reply crates/jstz_cli/src/utils.rs:13]: We can probably use a `Result` here. Will fix 👍

File: crates/jstz_cli/src/logs/trace.rs (hunk 3 above)
Comment (johnyob) [reply crates/jstz_cli/src/logs/trace.rs:38]: It should've been `Event source closed with an error: {}`. Will fix

File: crates/jstz_cli/src/sandbox/daemon.rs (hunk 2 above)
Comment (johnyob) [reply crates/jstz_cli/src/sandbox/daemon.rs]: Fixed 👍 (also changed the `sandbox stop` error to match)

File: crates/jstz_cli/src/logs/trace.rs (hunk 3 above)
Comment (johnyob) [reply crates/jstz_cli/src/logs/trace.rs]: Fixed

File: crates/jstz_cli/src/utils.rs (hunk 1 above)
Comment (johnyob) [reply crates/jstz_cli/src/utils.rs]: Fixed, it's slightly more messy. But it works :)

**Review by sam-finch-tezos — APPROVED** [review]

**Pull request merged**